File: app/src/main/java/com/hmmelton/kibbble/models/Pet.java
package com.hmmelton.kibbble.models;

import com.google.firebase.database.PropertyName;

import java.io.Serializable;
import java.util.List;

/**
 * Created by harrison on 4/15/17.
 * This is a model class representing a pet available for adoption.
 */

public class Pet implements Serializable {

    @PropertyName("id")
    String mId;
    @PropertyName("name")
    String mName;
    @PropertyName("breed")
    String mBreed;
    @PropertyName("age")
    int mAge;
    @PropertyName("saves")
    int mSaves;
    @PropertyName("gender")
    String mGender;
    @PropertyName("size")
    String mSize;
    @PropertyName("color")
    String mColor;
    @PropertyName("spayed_neutered")
    boolean mSpayedNeutered;
    @PropertyName("housetrained")
    boolean mHousetrained;
    @PropertyName("bio")
    String mBio;
    @PropertyName("images")
    List<String> mImages;
    @PropertyName("shelter_id")
    String mShelterId;

    // Empty constructor for Firebase
    public Pet() {}

    /* Getter methods */

    public String getId() {
        return mId;
    }

    public String getName() {
        return mName;
    }

    public String getBreed() {
        return mBreed;
    }

    public int getAge() {
        return mAge;
    }

    public int getSaves() {
        return mSaves;
    }

    public String getGender() {
        // TODO: change this in the future for localization
        return mGender.equals("male") ? "Male" : "Female";
    }

    public String getSize() {
        return mSize;
    }

    public String getColor() {
        return mColor;
    }

    public boolean isSpayedNeutered() {
        return mSpayedNeutered;
    }

    public boolean isHousetrained() {
        return mHousetrained;
    }

    public String getBio() {
        return mBio;
    }

    public List<String> getImages() {
        return mImages;
    }

    public String getShelterId() {
        return mShelterId;
    }
}

File: app/src/main/java/com/hmmelton/kibbble/models/Pet.kt
package com.hmmelton.kibbble.models

import com.google.firebase.database.PropertyName
import java.io.Serializable

/**
 * Created by harrisonmelton on 4/30/17.
 * This is a data model for pets.
 */
// Empty constructor for Firebase
class Pet : Serializable {

    /* Getter methods */

    @PropertyName("id")
    var id: String? = null
        internal set
    @PropertyName("name")
    var name: String? = null
        internal set get
    @PropertyName("breed")
    var breed: String? = null
        internal set
    @PropertyName("age")
    var age: Int = 0
        internal set
    @PropertyName("saves")
    var saves: Int = 0
        internal set
    @PropertyName("gender")
    internal var mGender: String? = null
    @PropertyName("size")
    var size: String? = null
        internal set
    @PropertyName("color")
    var color: String? = null
        internal set
    @PropertyName("spayed_neutered")
    var isSpayedNeutered: Boolean = false
        internal set
    @PropertyName("housetrained")
    var isHousetrained: Boolean = false
        internal set
    @PropertyName("bio")
    var bio: String? = null
        internal set
    @PropertyName("images")
    var images: List<String>? = null
        internal set
    @PropertyName("shelter_id")
    var shelterId: String? = null
        internal set

    // TODO: change this in the future for localization
    val gender: String
        get() = if (mGender == "male") "Male" else "Female"
}